packages/stats/gbstats/devtools/simulation.py

```diff
@@ -1,98 +1,95 @@
 ###############################################
-# Doesnt it seem like you should pass in some estimator class
-# that always returns a point estimate and a standard error,
-# and more or less everything else is handled by this simulation engine?
 
-# You basically need to have fixed interfaces across classes
-# (input: Statistic output: se and mean ). Which we kind of have.
 
+class SimulationStudy(ABC):
+    def __init__(
+        self,
+        test_dict: Mapping[str, Tuple[Type[BaseABTest], BaseConfig]],
+        data_params: Dict,
+        seed: int,
+        n_sim: int = 100,
+        alpha: float = 0.05,
+    ):
+        self.n_sim = n_sim
+        self.alpha = alpha
+        self.seed = seed
+        self.test_names = list(test_dict.keys())
+        self.tests = list([v[0] for v in test_dict.values()])
+        self.configs = list([v[1] for v in test_dict.values()])
+        self.data_params = data_params
+        self.create_storage_arrays()
 
-@dataclass
-class Estimator:
-    n_sim: int
-    alpha: float = 0.05
+    def run_sim(self):
+        for i in range(self.n_sim):
+            self.run_iteration(i)
+
+    def run_iteration(self, i):
+        np.random.seed(self.seed + i)
+        stat_a, stat_b, estimand = self.generate_data()
+        for j, test in enumerate(self.tests):
+            t = test(stat_a, stat_b, self.configs[j])
+            test_result = t.compute_result()
+            self.pt[i, j] = test_result.uplift.mean
+            self.se[i, j] = test_result.uplift.stddev
+            self.lower_limit[i, j] = test_result.ci[0]
+            self.upper_limit[i, j] = test_result.ci[1]
+            self.theta[i, j] = estimand
 
-    def create_arrays(self):
-        array_shape = (self.n_sim,)
+    def create_storage_arrays(self):
+        array_shape = (self.n_sim, len(self.tests))
         self.pt = np.empty(array_shape)
         self.se = np.empty(array_shape)
         self.theta = np.empty(array_shape)
+        self.lower_limit = np.empty(array_shape)
+        self.upper_limit = np.empty(array_shape)
 
-    @property
-    def multiplier(self):
-        return norm.ppf(1.0 - 0.5 * self.alpha, loc=0, scale=1)
-
-    @property
-    def lower_limit(self) -> np.ndarray:
-        return self.pt - self.se * self.multiplier
-
-    @property
-    def upper_limit(self) -> np.ndarray:
-        return self.pt + self.se * self.multiplier
+    @abstractmethod
+    def generate_data(self) -> Tuple[TestStatistic, TestStatistic, float]:
+        pass
 
     @property
-    def coverage(self) -> float:
+    def coverage(self):
         """computes coverage."""
-        return cast(
-            float,
+        return [
             np.mean(
-                (self.lower_limit <= self.theta) * (self.upper_limit >= self.theta)
-            ),
-        )
+                (self.lower_limit[:, j] <= self.theta[:, j])
+                * (self.upper_limit[:, j] >= self.theta[:, j])
+            )
+            for j in range(len(self.tests))
```

Comment:
I would create variable self.n_tests and use that throughout, rather than recalculating using len